### ISSUE TITLE
Crashing the app when importing PDF

### ISSUE THREAD

**Issue by MartinVoplakal** [saber-notes/saber#1385]
### Before submitting an issue, please check the following

- [X] I have searched for similar issues (both open and closed) and cannot find a duplicate
- [X] I agree to follow the [Code of Conduct](https://github.com/saber-notes/saber/blob/main/CODE_OF_CONDUCT.md)

### Describe the bug

When importing pdf about 700 pages the app freezes and crashes. I use this wonderful app on Samsung tablet. I uploaded mentioned pdf file here.

### To reproduce

Open the app
Try to import large pdf

### Expected behavior

Expect some loading not crashing 

### Saber version

0.25.1

### Device

- Device: samsung tablet 
- OS: android 13

**Comment by MartinVoplakal** [comment]
[Uploading MA1-ZS24 (1).pdf…]()

**Comment by superbm14** [comment]
For me, this got improved in Saber v25.2

**Comment by f-engg** [comment]
I have this on iPadOS, every time I try to open a vector file pdf in saber it crashes.
that same pdf works on Nebo and Goodnotes 5

**Comment by InfernalConcept** [comment]
Yes, it still crashes when importing bigger pdfs with a lot of pages.  May be it should not load the hole file just the preview and just the page you want to view and edit? Seems to fill up the RAM of weaker devices, loading for a long time and crashes. Device: Samsung galaxy S6 Lite with LineageOS (Android 15) with and F-droid Version 0.25.4 (250403)  seems to link the wrong arch (arm64 doesnt even start). Not usable for me atm.

**Comment by MelwinQ** [comment]
Similar behaviour in version v0.25.9 on Windows 11:
1) for some PDFs, import works flawlessly.
2) for some PDFs, the imported PDF is displayed, but the program crashes upon pressing the "Save" button. The Saber-file is created, though, only the thumbnail (preview) is missing. Reopening the imported note and resaving it creates the thumbnail.
3) for some PDFs, the app crashes immediately after choosing the PDF; no Saber files are created.

I could not detect any pattern when case 1, 2 or 3 occurs; there seems to be no relation to file size nor page number.
A file causing case 3 is attached.

On Samsung Tab 9+, Android 15, none of these errors occured.

[test_2page.pdf](https://github.com/user-attachments/files/20779596/test_2page.pdf)
